examples/IEKF_NavstateExample.cpp

```diff
@@ -0,0 +1,95 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file IEKF_NavstateExample.cpp
+ * @brief InvariantEKF on NavState (SE_2(3)) with IMU (predict) and GPS (update)
+ * @date April 25, 2025
+ * @authors Scott Baker, Matt Kielo, Frank Dellaert
+ */
+
+#include <gtsam/base/Matrix.h>
+#include <gtsam/base/OptionalJacobian.h>
+#include <gtsam/navigation/InvariantEKF.h>
+#include <gtsam/navigation/NavState.h>
+
+#include <iostream>
+
+using namespace std;
+using namespace gtsam;
+
+/**
+ * @brief Left-invariant dynamics for NavState.
+ * @param imu  6×1 vector [a; ω]: linear acceleration and angular velocity.
+ * @return     9×1 tangent: [ω; 0₃; a].
+ */
+Vector9 dynamics(const Vector6& imu) {

```

Comment:
@scottiyio I think this example still needs some unit-testing. In particular, show that
- $X \cdot U = X \cdot \exp(\delta t \xi^\wedge)$ the same as the nonlinear dynamics in Potokar
- Show that $Ad_{U^{-1}}$ matches the LIEKF Jacobian in Potokar.